rosidl_generator_rs/cmake/rosidl_generator_rs_generate_interfaces.cmake

```diff
@@ -162,93 +155,30 @@ set_property(
   ${_generated_extension_files}
   ${_generated_common_rs_files}
   ${_generated_msg_rs_files}
-  ${_generated_msg_c_files}
   ${_generated_srv_rs_files}
   PROPERTY GENERATED 1)
 
-set(_type_support_by_generated_c_files ${_type_support_by_generated_msg_c_files} ${_type_support_by_generated_srv_c_files})
-set(_generated_rs_files ${_generated_msg_rs_files} ${_generated_srv_rs_files})
-
 set(_rsext_suffix "__rsext")
 foreach(_typesupport_impl ${_typesupport_impls})
   find_package(${_typesupport_impl} REQUIRED)
 
-  set(_extension_compile_flags "")
-  if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    set(_extension_compile_flags -Wall -Wextra)
-  endif()
-
-  set(_target_name "${PROJECT_NAME}__${_typesupport_impl}${_rsext_suffix}")
-
-  add_library(${_target_name} SHARED
-    ${_generated_extension_${_typesupport_impl}_files}
-  )
-  add_dependencies(
-    ${_target_name}
-    ${rosidl_generate_interfaces_TARGET}${_target_suffix}
-    ${rosidl_generate_interfaces_TARGET}__rosidl_typesupport_c
-  )
-
-  target_link_libraries(
-    ${_target_name}
-    ${PROJECT_NAME}__${_typesupport_impl}
-    ${rosidl_generate_interfaces_TARGET}__rosidl_generator_c
-  )
-
-  if (NOT $ENV{ROS_DISTRO} STREQUAL "rolling")
-    rosidl_target_interfaces(${_target_name}
-      ${rosidl_generate_interfaces_TARGET} rosidl_typesupport_c)
-  else()
-    rosidl_get_typesupport_target(rust_typesupport_target ${rosidl_generate_interfaces_TARGET} rosidl_typesupport_c)
-    target_link_libraries(${_target_name} ${rust_typesupport_target})
-  endif()
-
-  target_include_directories(${_target_name}
-    PUBLIC
-    ${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_c
-    ${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_rs
-  )
-
-  ament_target_dependencies(${_target_name}
-    "rosidl_runtime_c"
-    "rosidl_typesupport_c"
-    "rosidl_typesupport_interface"
-  )
-  foreach(_pkg_name ${rosidl_generate_interfaces_DEPENDENCY_PACKAGE_NAMES})
-    ament_target_dependencies(${_target_name}
-      ${_pkg_name}
-    )
-  endforeach()
-
-  add_dependencies(${_target_name}
-    ${rosidl_generate_interfaces_TARGET}__${_typesupport_impl}
-  )
-  ament_target_dependencies(${_target_name}
-    "rosidl_runtime_c"
-    "rosidl_generator_rs"
-  )
-
   if(NOT rosidl_generate_interfaces_SKIP_INSTALL)
     install(
       DIRECTORY "${_output_path}/rust"
       DESTINATION "share/${PROJECT_NAME}"
     )
-    install(TARGETS ${_target_name}
-      ARCHIVE DESTINATION lib
-      LIBRARY DESTINATION lib
-    )
-
-    configure_file("${rosidl_generator_rs_TEMPLATE_DIR}/Cargo.toml.in"
-      "share/${PROJECT_NAME}/rust/Cargo.toml"
-      @ONLY)
-
-    install(
-      FILES "${CMAKE_CURRENT_BINARY_DIR}/share/${PROJECT_NAME}/rust/Cargo.toml"
-      DESTINATION share/${PROJECT_NAME}/rust/
-    )
   endif()
 endforeach()
 
+configure_file("${rosidl_generator_rs_TEMPLATE_DIR}/Cargo.toml.in"
```

Comment:
Not necessary to change this, but what do you think of using `.em` templates for `Cargo.toml` instead of CMake's limited template support?

Reply:
Sure, why not

Reply:
What does that mean? That you're going to switch to `.em` templates or not? Can you elaborate?

Reply:
I think it's a good idea, but I'd probably do it in the future instead of in this MR, since it's mostly unrelated.

Reply:
Yeah, that sounds good for a future PR